Assets/Scripts/Shell/BulletView.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TankBattle.Tank;
+
+namespace TankBattle.Bullet
+{
+    public class BulletView : MonoBehaviour
+    {
+        private BulletController controller;
+
+        public void SetController(BulletController _controller)
+        {
+            controller = _controller;
+        }
+
+        private void OnTriggerExit(Collider collider) 
+        {
+            if (collider.gameObject.tag == "GameBoundary")
+            {
+                if (controller != null)
+                {
+                    controller.DestroyBullet();
+                }
+            }
+        }
+
+        private void OnTriggerEnter(Collider collision)
+        {
+            if (collision.gameObject.tag == "Tank")
```

Comment:
no tag at all here - it should be using the interface

Reply:
I havnt learned interfaces yet. It's with state machines and im watching the tutorial. I'll implement it as soon as I complete it.

Reply:
cool mark this resolved then.